app/screens/TransactionAuthorization.tsx

```diff
@@ -80,43 +80,32 @@ export function TransactionAuthorization (): JSX.Element | null {
 
   const onRetry = useCallback(async (attempts: number) => {
     setIsRetry(true)
-    setPin('')
     setAttemptsRemaining(MAX_PASSCODE_ATTEMPT - attempts)
     await PasscodeAttemptCounter.set(attempts)
-    emitEvent('PIN')
   }, [attemptsRemaining])
 
-  // transaction signing specific callback
-  const onComplete = useCallback(async (dispatch: Dispatch<any>, tx: CTransactionSegWit) => {
-    setIsRetry(false)
-    setPin('')
+  const onPrompt = useCallback(async () => {
+    return await new Promise<string>((resolve, reject) => {
+      // passphrase prompt is meant for authorizing single transaction regardless
+      // caller should not prompt for next transaction before one is completed
+      // proxy the promise, wait for user input
+      PASSPHRASE_PROMISE_PROXY = {
+        resolve, reject
+      }
+      setPin('')
+      emitEvent('PIN')
+    })
+  }, [])
+
+  const resetPasscodeCounter = useCallback(async () => {
     setAttemptsRemaining(MAX_PASSCODE_ATTEMPT)
     await PasscodeAttemptCounter.set(0)
-
-    dispatch(transactionQueue.actions.pop()) // remove job
-    dispatch(ocean.actions.queueTransaction({ tx })) // push signed result for broadcasting
   }, [])
 
-  const onPrompt = useCallback(async () => {
-    if (status === 'PIN' || status === 'SIGNING') {
-      throw Error('Prompt UI occupied')
-    } else {
-      return await new Promise<string>((resolve, reject) => {
-        // passphrase prompt is meant for authorizing single transaction regardless
-        // caller should not prompt for next transaction before one is completed
-        // proxy the promise, wait for user input
-        PASSPHRASE_PROMISE_PROXY = {
-          resolve, reject
-        }
-        emitEvent('PIN')
-      })
-    }
-  }, [status])
```

Comment:
main issue, although this callback is aware (dep list) of `status` update. However, the instance of `onPrompt` passed/provided do use the initial value.

Adding `onPrompt` as dep list to useEffect will lead to cyclic dep child->parent->child->... infinite re-render.
Have to give up the extra check here.

The status check before start signing/authenticate is the guard to prevent start prompt for pin.